server/manager/src/main/java/org/apache/accumulo/master/recovery/RecoveryManager.java

```diff
@@ -148,6 +149,18 @@ private boolean exists(final Path path) throws IOException {
     }
   }
 
+  public String getHostPort() {
+
+    String hostPort = "";
+    Set<TServerInstance> tserverInstances = master.onlineTabletServers();
+
+    if (tserverInstances.isEmpty() && tserverInstances.size() < 0) {
+      return hostPort = null;
+
+    } else
+      return hostPort = tserverInstances.stream().findFirst().get().toString();
+  }
+
```

Comment:
Size can't be less than 0.

```suggestion
  public String getHostPort() {
    Set<TServerInstance> tserverInstances = master.onlineTabletServers();
    return tserverInstances.isEmpty() ? null : tserverInstances.stream().findFirst().get().toString();
  }

```

Reply:
Do you have an idea on how to properly implement this? `RecoveryManager.recoverLogs` gets called in `tabletGroupWatcher.run` when the tablets are not hosted and there are walogs. From testing, `tls.current` is null at this point. We could use `tls.last` but I am unsure if last will always be set and is prone to change.  Thoughts?

Reply:
@Manno15 If the question is directed at me: I don't, not without digging into this to come up with a solution myself.

Reply:
> @Manno15 If the question is directed at me: I don't, not without digging into this to come up with a solution myself. If tls.last isn't guaranteed to be the tserver that is going to be assigned this, and performing the recovery, then that would not be appropriate either.

I am not sure yet if tls.last is guranteed. I or @tynyttie  would have to do more research on a bigger cluster to get to the bottom of this. 

> it seems to me that the basic requirement is that the manager must decide the destination path, then somehow communicate that to the tserver that was assigned to do the sorting work. 

Based on what I am reading, the Tserver side of recover is only called in `Tablet.java`. This is after `RecoveryManager` is finishing do its side of te work. Not sure how everything fits in to get the full picture yet.